### PR TITLE
Issue #183:  Prevent health check apis from opentracing

### PR DESCRIPTION
**Pull request by razorao**
Added a filter to grpcopentracing's Interceptor to filter out the health check APIs

### CODE REVIEW

File: internal/server/server.go [hunk 1]
```diff
@@ -134,7 +134,13 @@ func newGrpcServer(r registerGrpcHandlers, interceptors ...grpc.UnaryServerInter
 		// Add tags to logger context.
 		grpcinterceptor.UnaryServerLoggerInterceptor(),
 		// Todo: Confirm tracing is working as expected. Jaegar integration?
-		grpcopentracing.UnaryServerInterceptor(),
+		grpcopentracing.UnaryServerInterceptor(grpcopentracing.WithFilterFunc(func(ctx context.Context, fullMethodName string) bool {
+			if fullMethodName == "/google.pubsub.v1.StatusCheckAPI/LivenessCheck" ||
```
Comment (shivasishdas) [inline internal/server/server.go:138]: better to maintain a running list of URLs as a constant somewhere. This list may/can expand in the future.

File: internal/server/server.go [hunk 2]
```diff
@@ -134,7 +134,13 @@ func newGrpcServer(r registerGrpcHandlers, interceptors ...grpc.UnaryServerInter
 		// Add tags to logger context.
 		grpcinterceptor.UnaryServerLoggerInterceptor(),
 		// Todo: Confirm tracing is working as expected. Jaegar integration?
-		grpcopentracing.UnaryServerInterceptor(),
+		grpcopentracing.UnaryServerInterceptor(grpcopentracing.WithFilterFunc(func(ctx context.Context, fullMethodName string) bool {
```
Comment (shivasishdas) [inline internal/server/server.go:137]: Explore possible UT?

**Review by shivasishdas — CHANGES_REQUESTED** [review]

File: internal/server/server.go [hunk 3]
```diff
@@ -196,3 +196,19 @@ func newInternalServer() (*http.Server, error) {
 
 	return &server, nil
 }
+
+// traceMethod - determines if opentracing should be enabled for the grpc method
+func traceMethod(ctx context.Context, fullMethodName string) bool {
+	donotTraceMethods := []string{ // Should be a very small list
```
Comment (shivasishdas) [inline internal/server/server.go:202]: nit: I understand the list is small for now, but still better to keep this as a map here for a constant time lookup.

File: internal/server/server.go [hunk 4]
```diff
@@ -196,3 +196,19 @@ func newInternalServer() (*http.Server, error) {
 
 	return &server, nil
 }
+
+// traceMethod - determines if opentracing should be enabled for the grpc method
+func traceMethod(ctx context.Context, fullMethodName string) bool {
```
Comment (shivasishdas) [inline internal/server/server.go:201]: a better name like `shouldEnableTrace`?

File: internal/server/server.go (hunk 3 above)
Comment (shivasishdas) [reply internal/server/server.go:202]: that being said, can this being a `var map[string]bool` init'ed on boot instead of a separate function?

**Review by shivasishdas — CHANGES_REQUESTED** [review]

**Review by sunnyagg — APPROVED** [review]

File: internal/server/server.go [hunk 5]
```diff
@@ -196,3 +208,9 @@ func newInternalServer() (*http.Server, error) {
 
 	return &server, nil
 }
+
+// shouldEnableTrace - determines if opentracing should be enabled for the grpc method
+func shouldEnableTrace(ctx context.Context, fullMethodName string) bool {
```
Comment (shivasishdas) [inline internal/server/server.go:213]: nit: Actually, the function determines which path to exclude from the trace. So rename to shouldDisableTrace or shouldSkipTrace

**Review by shivasishdas — CHANGES_REQUESTED** [review]

File: internal/server/server.go (hunk 5 above)
Comment (razorao) [reply internal/server/server.go:213]: Check the implementation of `UnaryServerInterceptor`
The filter method needs to return `false` to skip tracing
Hence.. Should enable trace? `false` = no

**Review by shivasishdas — APPROVED** [review]

**Comment by razorao** [comment]
@sunnyagg @shivasishdas  Have added the environment as part of the service name in tracer configuration.

**Review by shivasishdas — APPROVED** [review]

**Pull request merged**